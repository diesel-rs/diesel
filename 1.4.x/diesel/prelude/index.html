<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Re-exports important traits and types. Meant to be glob imported when using Diesel."><meta name="keywords" content="rust, rustlang, rust-lang, prelude"><title>diesel::prelude - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script id="default-settings" ></script><script src="../../storage.js"></script><script src="../../crates.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg"><link rel="alternate icon" type="image/png" href="../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu" role="button">&#9776;</div><a href='../../diesel/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><h2 class="location">Module prelude</h2><div class="sidebar-elems"><div class="block items"><ul><li><a href="#structs">Structs</a></li><li><a href="#traits">Traits</a></li></ul></div><div id="sidebar-vars" data-name="prelude" data-ty="mod" data-relpath="./"></div><script defer src="./sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu" title="themes"><img width="18" height="18" alt="Pick another theme!" src="../../brush.svg"></button><div id="theme-choices" role="menu"></div></div><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input"name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><button type="button" id="help-button" title="help">?</button><a id="settings-menu" href="../../settings.html" title="settings"><img width="18" height="18" alt="Change settings" src="../../wheel.svg"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="in-band">Module <a href="../index.html">diesel</a>::<wbr><a class="mod" href="#">prelude</a><button id="copy-path" onclick="copy_path(this)" title="copy path"><img src="../../clipboard.svg" width="19" height="18" alt="Copy item import" title="Copy item import to clipboard"></button></span><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/diesel/lib.rs.html#336-364" title="goto source code">[src]</a></span></h1><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Re-exports important traits and types. Meant to be glob imported when using Diesel.</p>
</div></details><h2 id="reexports" class="section-header"><a href="#reexports">Re-exports</a></h2>
<div class="item-table"><div class="item-left import-item"><code>pub use associations::<a class="trait" href="../associations/trait.Identifiable.html" title="trait diesel::associations::Identifiable">Identifiable</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use connection::<a class="trait" href="../connection/trait.Connection.html" title="trait diesel::connection::Connection">Connection</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use deserialize::<a class="trait" href="../deserialize/trait.Queryable.html" title="trait diesel::deserialize::Queryable">Queryable</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use expression::<a class="trait" href="../expression/trait.AppearsOnTable.html" title="trait diesel::expression::AppearsOnTable">AppearsOnTable</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use expression::<a class="trait" href="../expression/trait.BoxableExpression.html" title="trait diesel::expression::BoxableExpression">BoxableExpression</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use expression::<a class="trait" href="../expression/trait.Expression.html" title="trait diesel::expression::Expression">Expression</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use expression::<a class="trait" href="../expression/trait.IntoSql.html" title="trait diesel::expression::IntoSql">IntoSql</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use expression::<a class="trait" href="../expression/trait.SelectableExpression.html" title="trait diesel::expression::SelectableExpression">SelectableExpression</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use <a class="mod" href="../expression_methods/index.html" title="mod diesel::expression_methods">expression_methods</a>::*;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use query_dsl::<a class="trait" href="../query_dsl/trait.QueryDsl.html" title="trait diesel::query_dsl::QueryDsl">QueryDsl</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use query_dsl::<a class="trait" href="../query_dsl/trait.RunQueryDsl.html" title="trait diesel::query_dsl::RunQueryDsl">RunQueryDsl</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use query_source::<a class="trait" href="../query_source/trait.Column.html" title="trait diesel::query_source::Column">Column</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use query_source::<a class="trait" href="../query_source/trait.QuerySource.html" title="trait diesel::query_source::QuerySource">QuerySource</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use query_source::<a class="trait" href="../query_source/trait.Table.html" title="trait diesel::query_source::Table">Table</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use result::<a class="enum" href="../result/enum.ConnectionError.html" title="enum diesel::result::ConnectionError">ConnectionError</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use result::<a class="type" href="../result/type.ConnectionResult.html" title="type diesel::result::ConnectionResult">ConnectionResult</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use result::<a class="trait" href="../result/trait.OptionalExtension.html" title="trait diesel::result::OptionalExtension">OptionalExtension</a>;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use result::<a class="type" href="../result/type.QueryResult.html" title="type diesel::result::QueryResult">QueryResult</a>;</code></div><div class="item-right docblock-short"></div></div><h2 id="structs" class="section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="struct" href="struct.MysqlConnection.html" title="diesel::prelude::MysqlConnection struct">MysqlConnection</a></div><div class="item-right docblock-short"><p>A connection to a MySQL database. Connection URLs should be in the form
<code>mysql://[user[:password]@]host/database_name</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.PgConnection.html" title="diesel::prelude::PgConnection struct">PgConnection</a></div><div class="item-right docblock-short"><p>The connection string expected by <code>PgConnection::establish</code>
should be a PostgreSQL connection string, as documented at
<a href="https://www.postgresql.org/docs/9.4/static/libpq-connect.html#LIBPQ-CONNSTRING">https://www.postgresql.org/docs/9.4/static/libpq-connect.html#LIBPQ-CONNSTRING</a></p>
</div><div class="item-left module-item"><a class="struct" href="struct.SqliteConnection.html" title="diesel::prelude::SqliteConnection struct">SqliteConnection</a></div><div class="item-right docblock-short"><p>Connections for the SQLite backend. Unlike other backends, “connection URLs”
for SQLite are file paths, <a href="https://sqlite.org/uri.html">URIs</a>, or special
identifiers like <code>:memory:</code>.</p>
</div></div><h2 id="traits" class="section-header"><a href="#traits">Traits</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="trait" href="trait.BelongingToDsl.html" title="diesel::prelude::BelongingToDsl trait">BelongingToDsl</a></div><div class="item-right docblock-short"><p>Constructs a query that finds record(s) based on directional association with other record(s).</p>
</div><div class="item-left module-item"><a class="trait" href="trait.GroupedBy.html" title="diesel::prelude::GroupedBy trait">GroupedBy</a></div><div class="item-right docblock-short"><p>The <code>grouped_by</code> function groups records by their parent.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.Insertable.html" title="diesel::prelude::Insertable trait">Insertable</a></div><div class="item-right docblock-short"><p>Represents that a structure can be used to insert a new row into the
database. This is automatically implemented for <code>&amp;[T]</code> and <code>&amp;Vec&lt;T&gt;</code> for
inserting more than one record.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.JoinOnDsl.html" title="diesel::prelude::JoinOnDsl trait">JoinOnDsl</a></div><div class="item-right docblock-short"><p>Specify the <code>ON</code> clause for a join statement. This will override
any implicit <code>ON</code> clause that would come from <a href="../macro.joinable.html"><code>joinable!</code></a></p>
</div><div class="item-left module-item"><a class="trait" href="trait.JoinTo.html" title="diesel::prelude::JoinTo trait">JoinTo</a></div><div class="item-right docblock-short"><p>Indicates that two tables can be joined without an explicit <code>ON</code> clause.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.SaveChangesDsl.html" title="diesel::prelude::SaveChangesDsl trait">SaveChangesDsl</a></div><div class="item-right docblock-short"><p>Sugar for types which implement both <code>AsChangeset</code> and <code>Identifiable</code></p>
</div></div></section><section id="search" class="content hidden"></section><div id="rustdoc-vars" data-root-path="../../" data-current-crate="diesel" data-search-index-js="../../search-index.js" data-search-js="../../search.js"></div>
    <script src="../../main.js"></script>
</body></html>