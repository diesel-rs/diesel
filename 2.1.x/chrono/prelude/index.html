<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="A convenience module appropriate for glob imports (`use chrono::prelude::*;`)."><title>chrono::prelude - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-46f98efaafac5295.ttf.woff2,FiraSans-Regular-018c141bf0843ffd.woff2,FiraSans-Medium-8f9a781e4970d388.woff2,SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2,SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../static.files/rustdoc-081576b923113409.css"><meta name="rustdoc-vars" data-root-path="../../" data-static-root-path="../../static.files/" data-current-crate="chrono" data-themes="" data-resource-suffix="" data-rustdoc-version="1.79.0-nightly (e3181b091 2024-04-18)" data-channel="nightly" data-search-js="search-5b46ae87cc198d72.js" data-settings-js="settings-4313503d2e1961c2.js" ><script src="../../static.files/storage-e32f0c247825364d.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../static.files/main-20a3ad099b048cf2.js"></script><noscript><link rel="stylesheet" href="../../static.files/noscript-09095024cf37855e.css"></noscript><link rel="alternate icon" type="image/png" href="../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../chrono/index.html">chrono</a><span class="version">0.4.38</span></h2></div><h2 class="location"><a href="#">Module prelude</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#reexports">Re-exports</a></li></ul></section><h2><a href="../index.html">In crate chrono</a></h2></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><div id="sidebar-button" tabindex="-1"><a href="../../chrono/all.html" title="show sidebar"></a></div><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Type ‘S’ or ‘/’ to search, ‘?’ for more options…" type="search"><div id="help-button" tabindex="-1"><a href="../../help.html" title="help">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../settings.html" title="settings">Settings</a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../index.html">chrono</a>::<wbr><a class="mod" href="#">prelude</a><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><span class="out-of-band"><a class="src" href="../../src/chrono/lib.rs.html#527">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>A convenience module appropriate for glob imports (<code>use chrono::prelude::*;</code>).</p>
</div></details><h2 id="reexports" class="section-header">Re-exports<a href="#reexports" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name" id="reexport.Date"><code>pub use crate::<a class="struct" href="../struct.Date.html" title="struct chrono::Date">Date</a>;</code></div><div class="desc docblock-short"><span class="stab deprecated" title="">Deprecated</span></div></li><li><div class="item-name" id="reexport.Local"><code>pub use crate::<a class="struct" href="../offset/struct.Local.html" title="struct chrono::offset::Local">Local</a>;</code></div></li><li><div class="item-name" id="reexport.SubsecRound"><code>pub use crate::<a class="trait" href="../round/trait.SubsecRound.html" title="trait chrono::round::SubsecRound">SubsecRound</a>;</code></div></li><li><div class="item-name" id="reexport.DateTime"><code>pub use crate::<a class="struct" href="../struct.DateTime.html" title="struct chrono::DateTime">DateTime</a>;</code></div></li><li><div class="item-name" id="reexport.SecondsFormat"><code>pub use crate::<a class="enum" href="../format/enum.SecondsFormat.html" title="enum chrono::format::SecondsFormat">SecondsFormat</a>;</code></div></li><li><div class="item-name" id="reexport.Datelike"><code>pub use crate::<a class="trait" href="../trait.Datelike.html" title="trait chrono::Datelike">Datelike</a>;</code></div></li><li><div class="item-name" id="reexport.Month"><code>pub use crate::<a class="enum" href="../enum.Month.html" title="enum chrono::Month">Month</a>;</code></div></li><li><div class="item-name" id="reexport.Timelike"><code>pub use crate::<a class="trait" href="../trait.Timelike.html" title="trait chrono::Timelike">Timelike</a>;</code></div></li><li><div class="item-name" id="reexport.Weekday"><code>pub use crate::<a class="enum" href="../enum.Weekday.html" title="enum chrono::Weekday">Weekday</a>;</code></div></li><li><div class="item-name" id="reexport.FixedOffset"><code>pub use crate::<a class="struct" href="../offset/struct.FixedOffset.html" title="struct chrono::offset::FixedOffset">FixedOffset</a>;</code></div></li><li><div class="item-name" id="reexport.Utc"><code>pub use crate::<a class="struct" href="../offset/struct.Utc.html" title="struct chrono::offset::Utc">Utc</a>;</code></div></li><li><div class="item-name" id="reexport.NaiveDate"><code>pub use crate::<a class="struct" href="../naive/struct.NaiveDate.html" title="struct chrono::naive::NaiveDate">NaiveDate</a>;</code></div></li><li><div class="item-name" id="reexport.NaiveDateTime"><code>pub use crate::<a class="struct" href="../naive/struct.NaiveDateTime.html" title="struct chrono::naive::NaiveDateTime">NaiveDateTime</a>;</code></div></li><li><div class="item-name" id="reexport.NaiveTime"><code>pub use crate::<a class="struct" href="../naive/struct.NaiveTime.html" title="struct chrono::naive::NaiveTime">NaiveTime</a>;</code></div></li><li><div class="item-name" id="reexport.Offset"><code>pub use crate::<a class="trait" href="../trait.Offset.html" title="trait chrono::Offset">Offset</a>;</code></div></li><li><div class="item-name" id="reexport.TimeZone"><code>pub use crate::<a class="trait" href="../trait.TimeZone.html" title="trait chrono::TimeZone">TimeZone</a>;</code></div></li></ul></section></div></main></body></html>