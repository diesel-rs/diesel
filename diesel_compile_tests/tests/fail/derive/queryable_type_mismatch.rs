extern crate diesel;

use diesel::prelude::*;

table! {
    users {
        id -> Integer,
        name -> Text,
        bio -> Nullable<Text>,
    }
}

#[derive(Queryable)]
struct User {
    id: i32,
    name: String,
    bio: Option<String>,
}

#[derive(Queryable)]
struct UserWithToFewFields {
    id: i32,
    name: String,
}

#[derive(Queryable)]
struct UserWithToManyFields {
    id: i32,
    name: String,
    bio: Option<String>,
    age: i32,
}

#[derive(Queryable)]
struct UserWrongOrder {
    name: String,
    id: i32,
    bio: Option<String>,
}

#[derive(Queryable)]
struct UserTypeMismatch {
    id: i32,
    name: i32,
    bio: Option<String>,
}

#[derive(Queryable)]
struct UserNullableTypeMismatch {
    id: i32,
    name: Option<String>,
    bio: String,
}

fn test(conn: &mut PgConnection) {
    // check that this works fine
    let _ = users::table.load::<User>(conn);
    let _ = users::table.load::<UserWithToFewFields>(conn);
    //~^ ERROR: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied

    let _ = users::table.load::<UserWithToManyFields>(conn);
    //~^ ERROR: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied

    let _ = users::table.load::<UserWrongOrder>(conn);
    //~^ ERROR: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<UserWrongOrder, _>` is not satisfied

    let _ = users::table.load::<UserTypeMismatch>(conn);
    //~^ ERROR: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<UserTypeMismatch, _>` is not satisfied

    let _ = users::table.load::<UserNullableTypeMismatch>(conn);
    //~^ ERROR: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied
}

fn main() {}
