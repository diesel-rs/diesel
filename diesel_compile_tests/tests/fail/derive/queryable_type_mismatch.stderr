error[E0277]: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied
    --> tests/fail/derive/queryable_type_mismatch.rs:58:54
     |
58   |     let _ = users::table.load::<UserWithToFewFields>(conn);
     |                          ----                        ^^^^ the trait `load_dsl::private::CompatibleType<UserWithToFewFields, _>` is not implemented for `(Integer, Text, Nullable<Text>)`
     |                          |
     |                          required by a bound introduced by this call
     |
     = note: this is a mismatch between what your query returns and what your type expects the query to return
     = note: the fields in your struct need to match the fields returned by your query in count, order and type
     = note: consider using `#[diesel(check_for_backend(_))]` on either `#[derive(Selectable)]` or `#[derive(QueryableByName)]` 
             on your struct `UserWithToFewFields` and in your query `.select(UserWithToFewFields::as_select())` to get a better error message
     = help: the following other types implement trait `load_dsl::private::CompatibleType<U, DB>`:
               `Untyped` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::expression::select_by::SelectBy<U, DB>` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::sql_types::Nullable<(ST0, ST1)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5, ST6)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
             and N others
     = note: required for `users::table` to implement `LoadQuery<'_, _, UserWithToFewFields>`
note: required by a bound in `diesel::RunQueryDsl::load`
    --> DIESEL/diesel/diesel/src/query_dsl/mod.rs
     |
LL |     fn load<'query, U>(self, conn: &mut Conn) -> QueryResult<Vec<U>>
     |        ---- required by a bound in this associated function
LL |     where
LL |         Self: LoadQuery<'query, Conn, U>,
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `RunQueryDsl::load`
  
     
error[E0277]: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied
    --> tests/fail/derive/queryable_type_mismatch.rs:61:55
     |
61   |     let _ = users::table.load::<UserWithToManyFields>(conn);
     |                          ----                         ^^^^ the trait `load_dsl::private::CompatibleType<UserWithToManyFields, _>` is not implemented for `(Integer, Text, Nullable<Text>)`
     |                          |
     |                          required by a bound introduced by this call
     |
     = note: this is a mismatch between what your query returns and what your type expects the query to return
     = note: the fields in your struct need to match the fields returned by your query in count, order and type
     = note: consider using `#[diesel(check_for_backend(_))]` on either `#[derive(Selectable)]` or `#[derive(QueryableByName)]` 
             on your struct `UserWithToManyFields` and in your query `.select(UserWithToManyFields::as_select())` to get a better error message
     = help: the following other types implement trait `load_dsl::private::CompatibleType<U, DB>`:
               `Untyped` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::expression::select_by::SelectBy<U, DB>` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::sql_types::Nullable<(ST0, ST1)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5, ST6)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
             and N others
     = note: required for `users::table` to implement `LoadQuery<'_, _, UserWithToManyFields>`
note: required by a bound in `diesel::RunQueryDsl::load`
    --> DIESEL/diesel/diesel/src/query_dsl/mod.rs
     |
LL |     fn load<'query, U>(self, conn: &mut Conn) -> QueryResult<Vec<U>>
     |        ---- required by a bound in this associated function
LL |     where
LL |         Self: LoadQuery<'query, Conn, U>,
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `RunQueryDsl::load`
  
     
error[E0277]: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<UserWrongOrder, _>` is not satisfied
    --> tests/fail/derive/queryable_type_mismatch.rs:64:49
     |
64   |     let _ = users::table.load::<UserWrongOrder>(conn);
     |                          ----                   ^^^^ the trait `load_dsl::private::CompatibleType<UserWrongOrder, _>` is not implemented for `(Integer, Text, Nullable<Text>)`
     |                          |
     |                          required by a bound introduced by this call
     |
     = note: this is a mismatch between what your query returns and what your type expects the query to return
     = note: the fields in your struct need to match the fields returned by your query in count, order and type
     = note: consider using `#[diesel(check_for_backend(_))]` on either `#[derive(Selectable)]` or `#[derive(QueryableByName)]` 
             on your struct `UserWrongOrder` and in your query `.select(UserWrongOrder::as_select())` to get a better error message
     = help: the following other types implement trait `load_dsl::private::CompatibleType<U, DB>`:
               `Untyped` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::expression::select_by::SelectBy<U, DB>` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::sql_types::Nullable<(ST0, ST1)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5, ST6)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
             and N others
     = note: required for `users::table` to implement `LoadQuery<'_, _, UserWrongOrder>`
note: required by a bound in `diesel::RunQueryDsl::load`
    --> DIESEL/diesel/diesel/src/query_dsl/mod.rs
     |
LL |     fn load<'query, U>(self, conn: &mut Conn) -> QueryResult<Vec<U>>
     |        ---- required by a bound in this associated function
LL |     where
LL |         Self: LoadQuery<'query, Conn, U>,
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `RunQueryDsl::load`
  
     
error[E0277]: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<UserTypeMismatch, _>` is not satisfied
    --> tests/fail/derive/queryable_type_mismatch.rs:67:51
     |
67   |     let _ = users::table.load::<UserTypeMismatch>(conn);
     |                          ----                     ^^^^ the trait `load_dsl::private::CompatibleType<UserTypeMismatch, _>` is not implemented for `(Integer, Text, Nullable<Text>)`
     |                          |
     |                          required by a bound introduced by this call
     |
     = note: this is a mismatch between what your query returns and what your type expects the query to return
     = note: the fields in your struct need to match the fields returned by your query in count, order and type
     = note: consider using `#[diesel(check_for_backend(_))]` on either `#[derive(Selectable)]` or `#[derive(QueryableByName)]` 
             on your struct `UserTypeMismatch` and in your query `.select(UserTypeMismatch::as_select())` to get a better error message
     = help: the following other types implement trait `load_dsl::private::CompatibleType<U, DB>`:
               `Untyped` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::expression::select_by::SelectBy<U, DB>` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::sql_types::Nullable<(ST0, ST1)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5, ST6)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
             and N others
     = note: required for `users::table` to implement `LoadQuery<'_, _, UserTypeMismatch>`
note: required by a bound in `diesel::RunQueryDsl::load`
    --> DIESEL/diesel/diesel/src/query_dsl/mod.rs
     |
LL |     fn load<'query, U>(self, conn: &mut Conn) -> QueryResult<Vec<U>>
     |        ---- required by a bound in this associated function
LL |     where
LL |         Self: LoadQuery<'query, Conn, U>,
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `RunQueryDsl::load`
  
     
error[E0277]: the trait bound `(Integer, Text, Nullable<Text>): CompatibleType<..., _>` is not satisfied
    --> tests/fail/derive/queryable_type_mismatch.rs:70:59
     |
70   |     let _ = users::table.load::<UserNullableTypeMismatch>(conn);
     |                          ----                             ^^^^ the trait `load_dsl::private::CompatibleType<UserNullableTypeMismatch, _>` is not implemented for `(Integer, Text, Nullable<Text>)`
     |                          |
     |                          required by a bound introduced by this call
     |
     = note: this is a mismatch between what your query returns and what your type expects the query to return
     = note: the fields in your struct need to match the fields returned by your query in count, order and type
     = note: consider using `#[diesel(check_for_backend(_))]` on either `#[derive(Selectable)]` or `#[derive(QueryableByName)]` 
             on your struct `UserNullableTypeMismatch` and in your query `.select(UserNullableTypeMismatch::as_select())` to get a better error message
     = help: the following other types implement trait `load_dsl::private::CompatibleType<U, DB>`:
               `Untyped` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::expression::select_by::SelectBy<U, DB>` implements `load_dsl::private::CompatibleType<U, DB>`
               `diesel::sql_types::Nullable<(ST0, ST1)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
               `diesel::sql_types::Nullable<(ST0, ST1, ST2, ST3, ST4, ST5, ST6)>` implements `load_dsl::private::CompatibleType<Option<__T>, __DB>`
             and N others
     = note: required for `users::table` to implement `LoadQuery<'_, _, UserNullableTypeMismatch>`
note: required by a bound in `diesel::RunQueryDsl::load`
    --> DIESEL/diesel/diesel/src/query_dsl/mod.rs
     |
LL |     fn load<'query, U>(self, conn: &mut Conn) -> QueryResult<Vec<U>>
     |        ---- required by a bound in this associated function
LL |     where
LL |         Self: LoadQuery<'query, Conn, U>,
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `RunQueryDsl::load`
  
     For more information about this error, try `rustc --explain E0277`.
